src/session/session.module.js

```diff
@@ -0,0 +1,201 @@
+import angular from 'angular';
+import 'ngstorage';
+import querystring from 'query-string';
+
+
+function SessionFactory($http, $localStorage, $rootScope, $log, $state, $mdToast) {
+  return {
+    isAuthenticated() {
+      return !this.expired();
+    },
+
+    expired() {
+      if (!$localStorage.refresh_token) {
+        return true;
+      }
+      return this.expires() <= new Date();
+    },
+
+    expires() {
+      return new Date(JSON.parse($localStorage.refresh_token.exp) * 1000);
+    },
+
+    authorizationExpired() {
+      if (!$localStorage.authorization_token) {
+        return true;
+      }
+      return this.authorizationExpires() <= new Date();
```

Comment:
simplified syntax:

    return !$localStorage.authorization_token | this.authorizationExpires() <= new Date();